docs/source/content/handling_time_limits.md

```diff
@@ -0,0 +1,49 @@
+# Handling Time Limits
+In using Gym environments with reinforcement learning code, a common problem observed is how time limits are incorrectly handled. The `done` signal received from `env.step` indicates whether an episode has ended. However, this signal does not distinguish between the two different reasons why an episode can end - `termination` and `truncation`. 
+
+### Termination
+Termination refers to the episode ending after reaching a terminal state that is defined as part of the environment definition. Examples are - task success, task failure, robot falling down etc. Notably this also includes episode ending in finite-horizon environments due to a time-limit inherent to the environment. Note that to preserve Markov property, a representation of the remaining time must be present in the agent's observation in finite-horizon environments. [(Reference)](https://arxiv.org/abs/1712.00378)
+
+
+### Truncation
+Truncation refers to the episode ending after an externally defined time-limit. This time-limit is not part of the environment definition, and its sole purpose is practicality for the user collecting rollouts of the episode. 
```

Comment:
actually, truncation is wider than timelimit, for instance, for a robot, truncation would  also be going out of the tracking boundaries (e.g. quadruped locomotion).

But the treatment should be the same, in that example, you need to stop the episode (because you need to track your robot for computing reward/safety) but not punish the robot for that.

"is not part of the env definition" sounds weird because you actually usually define it there (in the `register()` method).

"sole purpose is practicality for the user collecting rollouts of the episode." -> this is quite vague...
And there are different reasons for timeouts:
- more diverse data collection (with different starting states)
- avoid infinite loop (for instance when using episodic RL or population based method)
- data at the beginning of an episode may be different than data after (for instance first lap for a racing car vs subsequent laps)

Reply:
> or a robot, truncation would also be going out of the tracking boundaries (e.g. quadruped locomotion).

I had never considered this, I will definitely add it